nx/test/nx_test.exs

```diff
@@ -1684,6 +1684,42 @@ defmodule NxTest do
     end
   end
 
+  describe "diag/2" do
+    test "extracts valid diagonal given no offset" do
```

Comment:
I think we need a test for the shape {m, n} when m > n and another for when m < n